implementations/rust/ockam/ockam_vault/src/traits/asymmetric_vault.rs

```diff
@@ -0,0 +1,87 @@
+use crate::{PublicKey, SecretAttributes, SmallBuffer};
+use ockam_core::{async_trait, compat::boxed::Box, KeyId, Result};
+
+/// Defines the Vault interface for asymmetric encryption.
+#[async_trait]
+pub trait AsymmetricVault: Send + Sync {
+    /// Compute and store an Elliptic-Curve Diffie-Hellman key using a secret key
+    /// and an uncompressed public key.
+    async fn ec_diffie_hellman(&self, secret: &KeyId, peer_public_key: &PublicKey)
+        -> Result<KeyId>;
+
+    /// Derive multiple output [`super::Secret`]s with given attributes using
+    /// the HKDF-SHA256 given the specified salt, info and input key
+    /// material.
+    async fn hkdf_sha256(
+        &self,
+        salt: &KeyId,
+        info: &[u8],
+        ikm: Option<&KeyId>,
+        output_attributes: SmallBuffer<SecretAttributes>,
+    ) -> Result<SmallBuffer<KeyId>>;
+}
+
+/// Tests for implementations of the AsymmetricVault trait
+pub mod tests {
```

Comment:
Can we avoid exposing it? Maybe hide by a feature flag, or move to a separate crate?

Reply:
These traits were already exposed in the previous version. But this is a good idea, I have added a feature flag.